pyriemann_qiskit/utils/distance.py

```diff
@@ -32,53 +50,100 @@ def logeucl_dist_convex(X, y, optimizer=ClassicalOptimizer()):
     References
     ----------
     .. [1] \
-        http://ibmdecisionoptimization.github.io/docplex-doc/mp/_modules/docplex/mp/model.html#Model
+        K. Zhao, A. Wiliem, S. Chen, and B. C. Lovell,
+        ‘Convex Class Model on Symmetric Positive Definite Manifolds’,
+        Image and Vision Computing, 2019.
+    .. [2] \
+        http://ibmdecisionoptimization.github.io/docplex-doc/cp/creating_model.html
+
     """
 
     optimizer = get_global_optimizer(optimizer)
 
-    n_classes, _, _ = X.shape
-    classes = range(n_classes)
+    n_matrices, _, _ = A.shape
+    matrices = range(n_matrices)
 
     def log_prod(m1, m2):
         return np.nansum(logm(m1).flatten() * logm(m2).flatten())
 
     prob = Model()
 
     # should be part of the optimizer
-    w = optimizer.get_weights(prob, classes)
+    w = optimizer.get_weights(prob, matrices)
 
-    _2VecLogYD = 2 * prob.sum(w[i] * log_prod(y, X[i]) for i in classes)
+    _2VecLogYD = 2 * prob.sum(w[i] * log_prod(B, A[i]) for i in matrices)
 
     wtDw = prob.sum(
-        w[i] * w[j] * log_prod(X[i], X[j]) for i in classes for j in classes
+        w[i] * w[j] * log_prod(A[i], A[j]) for i in matrices for j in matrices
     )
 
     objectives = wtDw - _2VecLogYD
 
     prob.set_objective("min", objectives)
 
-    result = optimizer.solve(prob, reshape=False)
+    prob.add_constraint(prob.sum(w) == 1)
 
-    return 1 - result
+    weights = optimizer.solve(prob, reshape=False)
+
+    # compute nearest matrix and distance
+    C = mean_logeuclid(A, weights)
+    distance = distance_logeuclid(C, B)
+
+    if return_weights:
+        return distance, weights
+    return distance
 
 
 _mdm_predict_distances_original = MDM._predict_distances
 
 
 def predict_distances(mdm, X):
-    if mdm.metric_dist == "convex":
+    if mdm.metric_dist == "logeuclid_cpm":
         centroids = np.array(mdm.covmeans_)
-        return np.array([logeucl_dist_convex(centroids, x) for x in X])
+
+        def postprocessed_distances(x):
+            _, weights = distance_logeuclid_cpm(centroids, x, return_weights=True)
+            return 1 - weights
+
+        return np.array([postprocessed_distances(x) for x in X])
```

Comment:
This should do the same job
```suggestion
        return 1 - np.array([logeucl_dist_convex(centroids, x) for x in X])
```

Reply:
You mean to reuse the deprecated function?

Reply:
Sorry, my suggestion is
```
weights = [distance_logeuclid_cpm(centroids, x, return_weights=True)[1] for x in X]
return 1 - np.array(weights)
```